### ISSUE TITLE
go1.18: [Tracker] go1.18 breaks CSR validation (fix pending in go1.18.1)

### ISSUE THREAD

**Issue by MadhavJivrajani** [kubernetes/kubernetes#108910]
### Which jobs are failing?

https://testgrid.k8s.io/sig-release-master-blocking#integration-master

### Which tests are failing?

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/108905/pull-kubernetes-integration/1506459579091783680
an instance of a failure, the tests here are the ones failing

### Since when has it been failing?

My guess is since https://github.com/kubernetes/kubernetes/pull/108870 was merged 

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#integration-master

### Reason for failure (if possible)

https://tip.golang.org/doc/go1.18#sha1

### Relevant SIG(s)

/sig testing

**Comment by nikhita** [comment]
> https://tip.golang.org/doc/go1.18#sha1

The failures for https://testgrid.k8s.io/sig-release-master-blocking#integration-master look like they are related to this.

[Example](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1506511365286662144) below:

```
x509: cannot verify signature: insecure algorithm SHA1-RSA (temporarily override with GODEBUG=x509sha1=1)
```

**Comment by aojea** [comment]
So,  presubmits jobs didn't catch these things?

**Comment by MadhavJivrajani** [comment]
A temporary fix, quick fix would be setting the env variable as suggested:
> GODEBUG=x509sha1=1

Continuing to look
/sig auth
(for visibility)

**Comment by MadhavJivrajani** [comment]
> So, presubmits jobs didn't catch these things?

yeah, I'm wondering this too

**Comment by aojea** [comment]
it seems golang has to do a minor release https://github.com/golang/go/issues/41682#issuecomment-1072695832, they are being to strict

**Comment by dims** [comment]
presubmits were using image based on golang 1.17, so we hit it AFTER the 1.18 in k/k merged and then the master image was updated from 1.17 to 1.18

**Comment by MadhavJivrajani** [comment]
The CI failures should now be fixed on account of: https://github.com/kubernetes/kubernetes/pull/108901
Thanks Dims!

**Comment by liggitt** [comment]
> we still need to fix things (regenerate hard-coded certs) in order to be fully back in business

Actually, I think the CSR failures are a go bug we need to wait for go1.18.1 for

**Comment by dims** [comment]
> it looks like all the integration test failures are for CSR objects, which will be resolved in go1.18.1
> 
> marking this as a blocker for 1.24

@liggitt do we have commitment from the golang team to ship go1.18.1 with the fix we need in time for 1.24? (what are our chances?)

**Comment by liggitt** [comment]
I don't know, but given it is a recovery release fixing a regression, I would expect it sooner than the normal ~month cadence

**Comment by dims** [comment]
Thanks @liggitt ! Looks like the backport issue is here: https://github.com/golang/go/issues/51852

**Comment by dims** [comment]
@MadhavJivrajani do you wanna open a CL in golang repo? https://github.com/golang/go/issues/51852#issuecomment-1086126145

**Comment by liggitt** [comment]
don't open a new CL, the one against master is under review and will be picked once ready

**Comment by dims** [comment]
@liggitt thanks. the linked comment caused me to wonder!

**Comment by liggitt** [comment]
golang master CL merged, pick open to release-1.18 https://go-review.googlesource.com/c/go/+/398074

**Comment by dims** [comment]
for this one ... i want to be proven wrong :) (i said, it may not go into 1.18.1 on record!)

**Comment by MadhavJivrajani** [comment]
Folks, Go 1.18.1 is postponed to 12th April. 
Please see: https://groups.google.com/g/golang-announce/c/vtbMjE04kPk/m/TY7AIrwSCgAJ?utm_medium=email&utm_source=footer
cc @palnabarun @dims

**Comment by pacoxu** [comment]
https://github.com/golang/go/releases/tag/go1.18.1 is out.
https://go.googlesource.com/go/+/refs/tags/go1.18.1

**Comment by pacoxu** [comment]
https://github.com/kubernetes/release/issues/2499 will update the dependency

**Comment by dims** [comment]
fixed k/k last remaining problem with 1.18.1 in https://github.com/kubernetes/kubernetes/pull/109312

/close
